### ISSUE TITLE
numpy 2.0 compatibility

### ISSUE THREAD

**Issue by ilyapopov** [PyDMD/PyDMD#558]
Currently, PyDMD is explicitly specified as incompatible with numpy 2.0 https://github.com/PyDMD/PyDMD/blob/1f6db1f15a6d07b435c6e6d25387c21f8c891074/pyproject.toml#L7

It would be nice if that limitation was lifted. 

Ref: https://numpy.org/doc/stable/numpy_2_0_migration_guide.html

**Comment by fathurwithyou** [comment]
Yes, I tried installing using pip and it showed error "ModuleNotFoundError: No module named 'numpy.char'" because of the numpy 1.26.4. To solve this problem, I upgrade the version to be 2.2.1. I think the lib should be updated.
